src/routes/RequestsRoute.js

```diff
@@ -634,6 +658,61 @@ class RequestsRoute extends React.Component {
     }
   }
 
+  toggleAllRows = () => {
+    const { resources } = this.props;
+    const { selectedRows } = this.state;
+
+    const toggledRows = resources.records.records.reduce((acc, row) => (
+      {
+        ...acc,
+        [row.id]: row,
+      }
+    ), {});
+
+    const filterSelectedRows = rows => {
+      Object.keys(toggledRows).forEach(id => {
+        if (rows[id]) delete rows[id];
+      });
+      return rows;
+    };
+
+    this.setState(({ selectedRows: this.getIsAllRowsSelected() ? filterSelectedRows(selectedRows) : { ...selectedRows, ...toggledRows } }));
+  };
+
+  getIsAllRowsSelected = () => {
+    const { resources } = this.props;
+    const { selectedRows } = this.state;
+    if (resources.records.records.length !== 0) {
+      const res = resources.records.records.every(({ id }) => Object.keys(selectedRows).includes(id));
+      return res;
```

Comment:
Could you simplify to:

```suggestion
      return resources.records.records.every(({ id }) => Object.keys(selectedRows).includes(id));
```